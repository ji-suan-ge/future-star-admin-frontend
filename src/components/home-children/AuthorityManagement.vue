<template>
  <el-container>
    <el-main>
      <el-row>
        <el-col :span="13" class="list-title">
          <h4>- 管理员列表 -</h4>
        </el-col>
        <el-col :span="4" offset="6" class="add-button-tar">
          <router-link to="/add-administrator" >
            <i class="el-icon-circle-plus-outline add-button"></i>
          </router-link>
        </el-col>
      </el-row>
      <el-row>
        <el-divider/>
        <el-table
          :data="tableData"
          v-loading="loading"
          id="table-center"
          stripe>
          <el-table-column
            fixed
            prop="name"
            align="center"
            label="姓名"
            :width="infoWidth">
          </el-table-column>
          <el-table-column
            prop="account"
            label="账号"
            align="center"
            width="70%">
          </el-table-column>
          <el-table-column
            label="招生权限"
            :width="privilegeWidth"
            align="center">
            <template slot-scope="scope">
              <el-switch
                v-model="scope.row.privilege.enrollment"
                :disabled="scope.row.privilege.super===1"
                :active-color="permissionActiveColor">
              </el-switch>
            </template>
          </el-table-column>
          <el-table-column
            label="课程权限"
            :width="privilegeWidth"
            align="center">
            <template slot-scope="scope">
              <el-switch
                v-model="scope.row.privilege.semester"
                :disabled="scope.row.privilege.super===1"
                :active-color="permissionActiveColor">
              </el-switch>
            </template>
          </el-table-column>
          <el-table-column
            label="校友活动权限"
            :width="privilegeWidth"
            align="center">
            <template slot-scope="scope">
              <el-switch
                v-model="scope.row.privilege.activity"
                :disabled="scope.row.privilege.super===1"
                :active-color="permissionActiveColor">
              </el-switch>
            </template>
          </el-table-column>
          <el-table-column
            label="校友资料权限"
            :width="privilegeWidth"
            align="center">
            <template slot-scope="scope">
              <el-switch
                v-model="scope.row.privilege.student"
                :disabled="scope.row.privilege.super===1"
                :active-color="permissionActiveColor">
              </el-switch>
            </template>
          </el-table-column>
          <el-table-column
            align="center"
            label="操作">
            <template slot-scope="scope">
              <el-button
                type="primary"
                size="small"
                @click="saveInfo(scope.row)"
                :disabled="scope.row.privilege.super===1"
                round>
                保存
              </el-button>
              <el-button
                type="danger"
                size="small"
                @click="deleteManager(scope.row, scope.$index)"
                :disabled="scope.row.privilege.super===1"
                round>
                删除
              </el-button>
            </template>
          </el-table-column>
        </el-table>
      </el-row>
      <el-row type="flex" justify="center">
        <el-pagination
          class="page-button"
          @current-change="CurrentPageChange"
          :current-page.sync="currentPage"
          :page-size="pageSize"
          layout="prev, pager, next, jumper"
          :total="total">
        </el-pagination>
      </el-row>
    </el-main>
  </el-container>
</template>

<script>
export default {
  name: 'AuthorityManagementPage',
  data () {
    return {
      permissionWidth: 200,
      accountWidth: 60,
      infoWidth: 130,
      privilegeWidth: 190,
      operationWidth: 142,
      permissionActiveColor: '#13ce66',
      pageSize: 10,
      currentPage: 1,
      total: 0,
      loading: true,
      tableData: [
      ]
    }
  },
  created () {
    let that = this
    if (that.axios) {
      that.getManagers()
    }
  },
  methods: {
    getManagers: function () {
      let that = this
      that.axios.get('/administrator/administrator', {
        params: {
          page: that.currentPage,
          page_size: that.pageSize
        }
      }).then(function (response) {
        if (response.data.code === '2000') {
          let administrators = response.data.data.results
          for (let i = 0; i < administrators.length; i++) {
            administrators[i].privilege.enrollment = administrators[i].privilege.enrollment === 1
            administrators[i].privilege.semester = administrators[i].privilege.semester === 1
            administrators[i].privilege.activity = administrators[i].privilege.activity === 1
            administrators[i].privilege.student = administrators[i].privilege.student === 1
          }
          that.total = response.data.data.count
          that.tableData = administrators
          that.loading = false
        } else {
          that.$message({
            type: 'error',
            message: '服务器繁忙，请稍后重试',
            duration: 2000
          })
        }
      }).catch(function (error) {
        console.log(error)
        that.$message({
          type: 'error',
          message: '服务器繁忙，请稍后重试',
          duration: 2000
        })
      })
    },
    deleteManager (administrator, index) {
      let that = this
      let url = '/administrator/administrator/' + administrator.id
      that.axios.delete(url
      ).then(function (response) {
        if (response.data.code === '2000') {
          that.tableData.splice(index, 1)
          that.$message({
            type: 'success',
            message: '删除成功',
            duration: 2000
          })
        } else {
          that.$message({
            type: 'error',
            message: '服务器繁忙，请稍后重试',
            duration: 2000
          })
        }
      }).catch(function (error) {
        console.log(error)
        that.$message({
          type: 'error',
          message: '服务器繁忙，请稍后重试',
          duration: 2000
        })
      })
    },
    saveInfo (administrator) {
      let that = this
      let url = '/administrator/administrator/' + administrator.id
      that.axios.put(url, {
        name: administrator.name,
        privilege: {
          enrollment: administrator.privilege.enrollment,
          semester: administrator.privilege.semester,
          activity: administrator.privilege.activity,
          student: administrator.privilege.student
        }
      }).then(function (response) {
        console.log(response)
        if (response.data.code === '2000') {
          that.$message({
            type: 'success',
            message: '修改成功',
            duration: 2000
          })
        } else {
          that.$message({
            type: 'error',
            message: '服务器繁忙，请稍后重试',
            duration: 2000
          })
        }
      }).catch(function (error) {
        console.log(error)
        that.$message({
          type: 'error',
          message: '服务器繁忙，请稍后重试',
          duration: 2000
        })
      })
    },
    CurrentPageChange: function (currentPage) {
      let that = this
      that.currentPage = currentPage
      that.getManagers()
    }
  }
}
</script>

<style scoped>
  .admin-table {
    margin: 40px 20px;
  }

  .page-button {
    margin-top: 2%;
  }

  .el-main {
    padding: 0;
    margin: 0;
  }

  .el-header {
    padding: 0;
    margin: 0;
  }

  .add-button-tar {
    margin-top: 2%;
    text-align: right;
  }

  .add-button {
    font-size: 30px;
    color: #272324;
  }

  .list-title {
    font-size: 25px;
    text-align: right;
  }

  h4 {
    color: #707070;
  }
</style>
