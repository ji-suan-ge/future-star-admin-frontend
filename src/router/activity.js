import CreateActivityPage from '../page/CreateActivityPage'
import ShowActivityParticipantsPage from '../page/ShowActivityParticipantsPage'
import EditExistingActivityPage from '../page/EditExistingActivityPage'
import ShowActivityInformationPage from '../page/ShowActivityInformationPage'

export default [
  {
    path: '/create-activity',
    name: 'CreateActivityPage',
    component: CreateActivityPage
  },
  {
    path: '/show-participants',
    name: 'ShowActivityParticipantsPage',
    component: ShowActivityParticipantsPage
  },
  {
    path: '/edit-activity',
    name: 'EditExistingActivityPage',
    component: EditExistingActivityPage
  },
  {
    path: '/show-activity-details',
    name: 'ShowActivityInformationPage',
    component: ShowActivityInformationPage
  }
]
