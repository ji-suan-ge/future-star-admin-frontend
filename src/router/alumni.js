import ShowAlumniInformationPage from '../page/alumni-management/ShowAlumniInformationPage'
import EditAlumniInformationPage from '../page/alumni-management/EditAlumniInformationPage'
import ShowAlumniEditHistoryPage from '../page/alumni-management/ShowAlumniEditHistoryPage'
import ShowAlumniHistoryInformationPage from '../page/alumni-management/ShowAlumniHistoryInformationPage'

export default [
  {
    path: '/show-all-information',
    name: 'ShowAlumniInformationPage',
    component: ShowAlumniInformationPage
  },
  {
    path: '/edit-alumni-information',
    name: 'EditAlumniInformationPage',
    component: EditAlumniInformationPage
  },
  {
    path: '/show-edit-history',
    name: 'ShowAlumniEditHistoryPage',
    component: ShowAlumniEditHistoryPage
  },
  {
    path: '/show-history-information',
    name: 'ShowAlumniHistoryInformationPage',
    component: ShowAlumniHistoryInformationPage
  }
]
