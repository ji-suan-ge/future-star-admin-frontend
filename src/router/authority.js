import AddAdministratorPage from '../page/authority-management/AddAdministratorPage'

export default [
  {
    path: '/add-administrator',
    name: 'AddAdministratorPage',
    component: AddAdministratorPage
  }
]
