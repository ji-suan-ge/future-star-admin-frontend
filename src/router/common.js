import LoginPage from '../page/LoginPage'
import HomePage from '../page/HomePage'
import homeChildren from './home-children'

export default [
  {
    path: '/',
    name: 'Root',
    redirect: '/login'
  },
  {
    path: '/login',
    name: 'LoginPage',
    component: LoginPage
  },
  {
    path: '/home',
    name: 'HomePage',
    component: HomePage,
    redirect: '/home/welcome',
    children: [
      ...homeChildren
    ]
  }
]
