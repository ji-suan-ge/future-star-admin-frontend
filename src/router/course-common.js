import DetailOfSemesterPage from '../page/course-common/DetailOfSemesterPage'
import DetailOfClassPage from '../page/course-common/DetailOfClassPage'
import DetailOfCoursePage from '../page/course-common/DetailOfCoursePage'
import DetailOfTeacherPage from '../page/course-common/DetailOfTeacherPage'
import DetailOfSubCoursePage from '../page/course-common/DetailOfSubCoursePage'
import EditSemesterInfoPage from '../page/course-common/EditSemesterInfoPage'
import EditClassInfoPage from '../page/course-common/EditClassInfoPage'
import EditCourseInfoPage from '../page/course-common/EditCourseInfoPage'
import EditTeacherInfoPage from '../page/course-common/EditTeacherInfoPage'
import AddCoursePage from '../page/course-common/AddCoursePage'
import AddCourseResourcePage from '../page/course-common/AddCourseResourcePage'
import StudentInClass from '../page/course-common/StudentInClass'

export default [
  {
    path: '/semester-detail',
    name: 'DetailOfSemesterPage',
    component: DetailOfSemesterPage
  },
  {
    path: '/class-detail',
    name: 'DetailOfClassPage',
    component: DetailOfClassPage
  },
  {
    path: '/course-detail',
    name: 'DetailOfCoursePage',
    component: DetailOfCoursePage
  },
  {
    path: '/teacher-detail',
    name: 'DetailOfTeacherPage',
    component: DetailOfTeacherPage
  },
  {
    path: '/sub-course-detail',
    name: 'DetailOfSubCoursePage',
    component: DetailOfSubCoursePage
  },
  {
    path: '/edit-semester-info',
    name: 'EditSemesterInfoPage',
    component: EditSemesterInfoPage
  },
  {
    path: '/edit-class-info',
    name: 'EditClassInfoPage',
    component: EditClassInfoPage
  },
  {
    path: '/edit-course-info',
    name: 'EditCourseInfoPage',
    component: EditCourseInfoPage
  },
  {
    path: '/edit-teacher-info',
    name: 'EditTeacherInfoPage',
    component: EditTeacherInfoPage
  },
  {
    path: '/add-course',
    name: 'AddCoursePage',
    component: AddCoursePage
  },
  {
    path: '/add-course-resource',
    name: 'AddCourseResourcePage',
    component: AddCourseResourcePage
  },
  {
    path: '/student-in-class',
    name: 'StudentInClass',
    component: StudentInClass
  }
]
