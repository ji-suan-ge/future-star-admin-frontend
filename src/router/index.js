import Vue from 'vue'
import VueRouter from 'vue-router'
import LoginPage from '../page/LoginPage'
import HomePage from '../page/HomePage'
import EnrollmentManagementPage from '../page/EnrollmentManagementPage'
import CourseManagementPage from '../page/CourseManagementPage'
import ActivityManagementPage from '../page/ActivityManagementPage'
import AlumniManagementPage from '../page/AlumniManagementPage'
import AuthorityManagementPage from '../page/AuthorityManagementPage'
import CreateActivityPage from '../page/CreateActivityPage'
import AddAdministrator from '../page/AddAdministrator'
import WelcomePage from '../page/WelcomPage'

Vue.use(VueRouter)

const routes = [
  {
    path: '/',
    name: 'Root',
    redirect: '/login'
  },
  {
    path: '/login',
    name: 'LoginPage',
    component: LoginPage
  },
  {
    path: '/home',
    name: 'HomePage',
    component: HomePage,
    redirect: '/home/welcome',
    children: [
      {
        path: 'welcome',
        name: 'WelcomePage',
        component: WelcomePage
      },
      {
        path: 'enrollment-management',
        name: 'EnrollmentManagementPage',
        component: EnrollmentManagementPage
      },
      {
        path: 'course-management',
        name: 'CourseManagementPage',
        component: CourseManagementPage
      },
      {
        path: 'activity-management',
        name: 'ActivityManagementPage',
        component: ActivityManagementPage
      },
      {
        path: 'alumni-management',
        name: 'AlumniManagementPage',
        component: AlumniManagementPage
      },
      {
        path: 'authority-management',
        name: 'AuthorityManagementPage',
        component: AuthorityManagementPage
      },
      {
        path: 'add-administrator',
        name: 'AddAdministrator',
        component: AddAdministrator
      }
    ]
  },
  {
    path: '/create-activity',
    name: 'CreateActivityPage',
    component: CreateActivityPage
  }
]

const router = new VueRouter({
  routes
})

export default router
