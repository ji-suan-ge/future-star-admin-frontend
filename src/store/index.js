import Vue from 'vue'
import Vuex from 'vuex'

Vue.use(Vuex)

export default new Vuex.Store({
  state: {
    activeIndexOfNavigation: '0',
    activeNameOfCourse: 'first',
    activeNameOfEnrollment: 'first',
    activeNameOfActivity: 'first',
    activeSemesterOfCourse: '',
    semesters: '',
    currentSemester: '',
    classes: '',
    currentClass: '',
    courses: '',
    currentCourse: '',
    infoOfAnAlumni: null
  },
  mutations: {
    changeActiveIndexOfNavigation (state, activeIndex) {
      state.activeIndexOfNavigation = activeIndex
    },
    changeActiveNameOfEnrollment (state, activeName) {
      state.activeNameOfEnrollment = activeName
    },
    changeActiveNameOfCourse (state, activeName) {
      state.activeNameOfCourse = activeName
    },
    changeActiveNameOfActivity (state, activeName) {
      state.activeNameOfActivity = activeName
    },
    changeActiveSemesterOfCourse (state, activeID) {
      state.activeSemesterOfCourse = activeID
    },
    changeSemesters (state, semesters) {
      state.semesters = semesters
    },
    changeInfoOfAnAlumni (state, alumniInfo) {
      state.infoOfAnAlumni = alumniInfo
    },
    changeCurrentSemester (state, semester) {
      state.currentSemester = semester
    },
    changeClasses (state, classes) {
      state.classes = classes
    },
    changeCurrentClass (state, currentClass) {
      state.currentClass = currentClass
    },
    changeCourses (state, courses) {
      state.courses = courses
    },
    changeCurrentCourse (state, currentCourse) {
      state.currentCourse = currentCourse
    },
    initializeStateWhileLogin (state) {
      state.activeIndexOfNavigation = '0'
      state.activeNameOfCourse = 'first'
      state.activeNameOfEnrollment = 'first'
      state.activeNameOfActivity = 'first'
      state.activeSemesterOfCourse = ''
      state.semesters = ''
      state.currentSemester = ''
      state.classes = ''
      state.currentClass = ''
      state.courses = ''
      state.currentCourse = ''
      state.infoOfAnAlumni = null
    },
    initializeStateWhileSwitch (state) {
      state.activeNameOfCourse = 'first'
      state.activeNameOfEnrollment = 'first'
      state.activeNameOfActivity = 'first'
      state.activeSemesterOfCourse = ''
      state.semesters = ''
      state.currentSemester = ''
      state.classes = ''
      state.currentClass = ''
      state.courses = ''
      state.currentCourse = ''
      state.infoOfAnAlumni = null
    }
  },
  actions: {
    changeActiveIndexOfNavigation (context, activeIndex) {
      context.commit('changeActiveIndexOfNavigation', activeIndex)
    },
    changeActiveNameOfEnrollment (context, activeName) {
      context.commit('changeActiveNameOfEnrollment', activeName)
    },
    changeActiveNameOfCourse (context, activeName) {
      context.commit('changeActiveNameOfCourse', activeName)
    },
    changeActiveNameOfActivity (context, activeName) {
      context.commit('changeActiveNameOfActivity', activeName)
    },
    changeActiveSemesterOfCourse (context, activeID) {
      context.commit('changeActiveSemesterOfCourse', activeID)
    },
    changeSemesters (context, semesters) {
      context.commit('changeSemesters', semesters)
    },
    changeInfoOfAnAlumni (context, alumniInfo) {
      context.commit('changeInfoOfAnAlumni', alumniInfo)
    },
    changeCurrentSemester (context, semester) {
      context.commit('changeCurrentSemester', semester)
    },
    changeClasses (context, classes) {
      context.commit('changeClasses', classes)
    },
    changeCurrentClass (context, currentClass) {
      context.commit('changeCurrentClass', currentClass)
    },
    changeCourses (context, courses) {
      context.commit('changeCourses', courses)
    },
    changeCurrentCourse (context, currentCourse) {
      context.commit('changeCurrentCourse', currentCourse)
    },
    initializeStateWhileLogin (context) {
      context.commit('initializeStateWhileLogin')
    },
    initializeStateWhileSwitch (context) {
      context.commit('initializeStateWhileSwitch')
    }
  },
  getters: {
    getActiveIndexOfNavigation (state) {
      return state.activeIndexOfNavigation
    },
    getActiveNameOfEnrollment (state) {
      return state.activeNameOfEnrollment
    },
    getActiveNameOfCourse (state) {
      return state.activeNameOfCourse
    },
    getActiveNameOfActivity (state) {
      return state.activeNameOfActivity
    },
    getActiveSemesterOfCourse (state) {
      return state.activeSemesterOfCourse
    },
    getSemesters (state) {
      return state.semesters
    },
    getInfoOfAnAlumni (state) {
      return state.infoOfAnAlumni
    },
    getCurrentSemester (state) {
      return state.currentSemester
    },
    getClasses (state) {
      return state.classes
    },
    getCurrentClass (state) {
      return state.currentClass
    },
    getCourses (state) {
      return state.courses
    },
    getCurrentCourse (state) {
      return state.currentCourse
    }
  },
  modules: {
  }
})
