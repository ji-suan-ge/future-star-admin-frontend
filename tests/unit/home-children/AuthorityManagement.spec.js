import { createLocalVue, mount } from '@vue/test-utils'
import AuthorityManagement from '../../../src/components/home-children/AuthorityManagement'
import Vuex from 'vuex'
import ElementUI from 'element-ui'
import VueRouter from 'vue-router'
import store from '../../../src/store'
import router from '../../../src/router'

describe('AuthorityManagement', () => {
  const localVue = createLocalVue()
  localVue.use(Vuex)
  localVue.use(ElementUI)
  localVue.use(VueRouter)
  const wrapper = mount(AuthorityManagement, {
    store,
    router,
    localVue
  })
  test('快照测试', async () => {
    await expect(wrapper.html()).toMatchSnapshot()
  })
  test('是一个 Vue 实例', async () => {
    await expect(wrapper.isVueInstance()).toBeTruthy()
  })
})
