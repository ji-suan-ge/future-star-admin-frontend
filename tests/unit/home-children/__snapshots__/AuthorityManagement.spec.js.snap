// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorityManagement 快照测试 1`] = `
<section class="el-container">
  <main class="el-main">
    <div class="el-row">
      <div class="list-title el-col el-col-13">
        <h4>- 管理员列表 -</h4>
      </div>
      <div class="add-button-tar el-col el-col-4 el-col-offset-6"><a href="#/add-administrator" class=""><i class="el-icon-circle-plus-outline add-button"></i></a></div>
    </div>
    <div class="el-row">
      <div class="el-divider el-divider--horizontal">
        <!---->
      </div>
      <div class="el-table el-table--fit el-table--striped el-table--scrollable-x el-loading-parent--relative" id="table-center">
        <div class="hidden-columns">
          <div></div>
          <div></div>
          <div></div>
          <div></div>
          <div></div>
          <div></div>
          <div></div>
        </div>
        <div class="el-table__header-wrapper">
          <table cellspacing="0" cellpadding="0" border="0" class="el-table__header" style="width: 1040px;">
            <colgroup>
              <col name="el-table_1_column_1" width="130">
              <col name="el-table_1_column_2" width="70">
              <col name="el-table_1_column_3" width="190">
              <col name="el-table_1_column_4" width="190">
              <col name="el-table_1_column_5" width="190">
              <col name="el-table_1_column_6" width="190">
              <col name="el-table_1_column_7" width="80">
            </colgroup>
            <thead class="">
              <tr class="">
                <th colspan="1" rowspan="1" class="el-table_1_column_1  is-center   is-hidden is-leaf">
                  <div class="cell">姓名</div>
                </th>
                <th colspan="1" rowspan="1" class="el-table_1_column_2  is-center   is-leaf">
                  <div class="cell">账号</div>
                </th>
                <th colspan="1" rowspan="1" class="el-table_1_column_3  is-center   is-leaf">
                  <div class="cell">招生权限</div>
                </th>
                <th colspan="1" rowspan="1" class="el-table_1_column_4  is-center   is-leaf">
                  <div class="cell">课程权限</div>
                </th>
                <th colspan="1" rowspan="1" class="el-table_1_column_5  is-center   is-leaf">
                  <div class="cell">校友活动权限</div>
                </th>
                <th colspan="1" rowspan="1" class="el-table_1_column_6  is-center   is-leaf">
                  <div class="cell">校友资料权限</div>
                </th>
                <th colspan="1" rowspan="1" class="el-table_1_column_7  is-center   is-leaf">
                  <div class="cell">操作</div>
                </th>
              </tr>
            </thead>
          </table>
        </div>
        <div class="el-table__body-wrapper is-scrolling-left">
          <table cellspacing="0" cellpadding="0" border="0" class="el-table__body" style="width: 1040px;">
            <colgroup>
              <col name="el-table_1_column_1" width="130">
              <col name="el-table_1_column_2" width="70">
              <col name="el-table_1_column_3" width="190">
              <col name="el-table_1_column_4" width="190">
              <col name="el-table_1_column_5" width="190">
              <col name="el-table_1_column_6" width="190">
              <col name="el-table_1_column_7" width="80">
            </colgroup>
            <tbody>
              <!---->
            </tbody>
          </table>
          <div class="el-table__empty-block" style="height: 100%; width: 1040px;"><span class="el-table__empty-text">暂无数据</span></div>
          <!---->
        </div>
        <!---->
        <div class="el-table__fixed" style="width: 130px;">
          <div class="el-table__fixed-header-wrapper">
            <table cellspacing="0" cellpadding="0" border="0" class="el-table__header" style="width: 1040px;">
              <colgroup>
                <col name="el-table_1_column_1" width="130">
                <col name="el-table_1_column_2" width="70">
                <col name="el-table_1_column_3" width="190">
                <col name="el-table_1_column_4" width="190">
                <col name="el-table_1_column_5" width="190">
                <col name="el-table_1_column_6" width="190">
                <col name="el-table_1_column_7" width="80">
              </colgroup>
              <thead class="">
                <tr class="">
                  <th colspan="1" rowspan="1" class="el-table_1_column_1  is-center   is-leaf">
                    <div class="cell">姓名</div>
                  </th>
                  <th colspan="1" rowspan="1" class="el-table_1_column_2  is-center   is-hidden is-leaf">
                    <div class="cell">账号</div>
                  </th>
                  <th colspan="1" rowspan="1" class="el-table_1_column_3  is-center   is-hidden is-leaf">
                    <div class="cell">招生权限</div>
                  </th>
                  <th colspan="1" rowspan="1" class="el-table_1_column_4  is-center   is-hidden is-leaf">
                    <div class="cell">课程权限</div>
                  </th>
                  <th colspan="1" rowspan="1" class="el-table_1_column_5  is-center   is-hidden is-leaf">
                    <div class="cell">校友活动权限</div>
                  </th>
                  <th colspan="1" rowspan="1" class="el-table_1_column_6  is-center   is-hidden is-leaf">
                    <div class="cell">校友资料权限</div>
                  </th>
                  <th colspan="1" rowspan="1" class="el-table_1_column_7  is-center   is-hidden is-leaf">
                    <div class="cell">操作</div>
                  </th>
                </tr>
              </thead>
            </table>
          </div>
          <div class="el-table__fixed-body-wrapper" style="top: 0px;">
            <table cellspacing="0" cellpadding="0" border="0" class="el-table__body" style="width: 1040px;">
              <colgroup>
                <col name="el-table_1_column_1" width="130">
                <col name="el-table_1_column_2" width="70">
                <col name="el-table_1_column_3" width="190">
                <col name="el-table_1_column_4" width="190">
                <col name="el-table_1_column_5" width="190">
                <col name="el-table_1_column_6" width="190">
                <col name="el-table_1_column_7" width="80">
              </colgroup>
              <tbody>
                <!---->
              </tbody>
            </table>
            <!---->
          </div>
          <!---->
        </div>
        <!---->
        <!---->
        <div class="el-table__column-resize-proxy" style="display: none;"></div>
        <div class="el-loading-mask el-loading-fade-enter el-loading-fade-enter-active" style="">
          <div class="el-loading-spinner"><svg viewBox="25 25 50 50" class="circular">
              <circle cx="50" cy="50" r="20" fill="none" class="path"></circle>
            </svg>
            <!---->
          </div>
        </div>
      </div>
    </div>
    <div class="el-row is-justify-center el-row--flex">
      <div class="page-button el-pagination"><button type="button" disabled="disabled" class="btn-prev"><i class="el-icon el-icon-arrow-left"></i></button>
        <ul class="el-pager">
          <li class="number active">1</li>
          <!---->
          <!---->
          <!---->
        </ul><button type="button" disabled="disabled" class="btn-next"><i class="el-icon el-icon-arrow-right"></i></button><span class="el-pagination__jump">前往<div class="el-input el-pagination__editor is-in-pagination"><!----><input type="number" autocomplete="off" min="1" max="1" class="el-input__inner"><!----><!----><!----><!----></div>页</span>
      </div>
    </div>
  </main>
</section>
`;
